gtk/src/callbacks.c

```diff
@@ -119,7 +119,7 @@ ghb_get_session_dbus_proxy(const gchar *name, const gchar *path, const gchar *in
 
     g_debug("ghb_get_session_dbus_proxy()");
     conn = g_bus_get_sync(G_BUS_TYPE_SESSION, NULL, &error);
-    if (conn == NULL)
+    if (error != NULL)
```

Comment:
error != NULL does not necessarily imply conn == NULL.  This needs to first check for conn == NULL, then check error != NULL before using error.

The rest looks good, thanks.

Reply:
I thought "Returns a GDBusConnection or NULL if error is set" means that.

Reply:
Convention is that you check for an error first (i.e. conn == NULL) then you use the the error code to give meaningful feedback to the user (if error code is set). 

Coders write terrible documentation, so it's a good idea to keep common conventions in mind when interpreting documentation *and* read the code to verify documentation whenever feasible :wink.  There are definitely cases in the code where both error and conn are returned with NULL values simultaneously.

I could easily see someone thinking it's a good idea to toss a success code into "error" and then forget to update documentation.